src/sentry/static/sentry/app/views/organizationStream/overview.jsx

```diff
@@ -317,6 +320,18 @@ const OrganizationStream = createReactClass({
 
         const queryCount = jqXHR.getResponseHeader('X-Hits');
         const queryMaxCount = jqXHR.getResponseHeader('X-Max-Hits');
+        const pageLinks = jqXHR.getResponseHeader('Link');
+        let initialCursor = this.state.initialCursor;
+
+        // Reset `initialCursor` if current route does not have "cursor" url param
+        // i.e. if you page back to first page and cursor is removed so that we fetch
+        // a fresh set of results, we should reset initial cursor
+        if ((!this.state.initialCursor || !currentQuery.cursor) && pageLinks) {
+          const links = utils.parseLinkHeader(pageLinks);
+          if (!links.previous.results) {
+            initialCursor = links.next.cursor;
```

Comment:
If you are on the 0th page won't this point at the 1st page?

Reply:
It does, we then use this cursor to find when we go from page 1 back to page 0, and when we do, we remove the cursor so it fetches an up-to-date set of results.

Reply:
Right. I didn't fully grok the other block the first time around.

Reply:
This assumption doesn't quite hold up when you navigate several pages deep. e.g. going from page 0 -> 1 -> 2 -> 1 -> 0, the cursor pointing from 0 -> 1 is not the same from 1 -> 0. I'm going to change this to just track a "page" number throughout.